Lib/test/test_py_compile.py

```diff
@@ -98,6 +98,18 @@ def test_bad_coding(self):
         self.assertFalse(os.path.exists(
             importlib.util.cache_from_source(bad_coding)))
 
+    def test_source_date_epoch(self):
+        testtime = 123456789
+        with support.EnvironmentVarGuard() as env:
+            env["SOURCE_DATE_EPOCH"] = str(testtime)
+            py_compile.compile(self.source_path, self.pyc_path)
+        self.assertTrue(os.path.exists(self.pyc_path))
+        self.assertFalse(os.path.exists(self.cache_path))
+        with open(self.pyc_path, "rb") as f:
```

Comment:
I would look at https://github.com/python/cpython/blob/9f1b7b93f5f0ef589e7b272e127cacf4ce5d23f1/Lib/test/test_py_compile.py#L125 as an example on how to make this test a little bit more robust in the face of `.pyc` files changing.